app/javascript/controllers/companion_window_controller.js

```diff
@@ -71,16 +71,17 @@ export default class extends Controller {
     if (!clickWithinDialog) modal.close()
   }
 
-  openShareModal() {
-    this.shareModalTarget.showModal()
+  openModal(event) {
+    const clickedButton = event.currentTarget.closest('button');
+    this.setReturnFocusButton(clickedButton);
+    const modalName = clickedButton.dataset['target'];
+    this[`${modalName}Target`].showModal()
   }
 
-  openDownloadModal() {
-    this.downloadModalTarget.showModal()
-  }
-
-  displayAccessibility() {
-    this.accessibilityTarget.showModal()
+  setReturnFocusButton(button) {
+    // we want the focus to return to share/download button not to the closed dropdown menu
+    if (button.parentElement.tagName == 'DIALOG') return;
+    this.returnFocusButton = button.matches(':focus-visible') ? button : false;
```

Comment:
I don't like mixing types here.  Can it  be

```suggestion
    if (button.parentElement.tagName !== 'DIALOG' && button.matches(':focus-visible'))
        this.returnFocusButton = button
```